QueryEngine/NativeCodegen.cpp

```diff
@@ -1077,6 +1081,101 @@ Executor::CompilationResult Executor::compileWorkUnit(const std::vector<InputTab
     }
   }
 
+  std::vector<llvm::Value*> hoisted_literals;
+
+  if (co.hoist_literals_ && !cgen_state_->defined_literals_.empty()) {
+    // we have some literals...
+
+    // create a new row_function with the literals as argument
+    std::vector<llvm::Type*> row_process_arg_types;
+
+    for (llvm::Function::arg_iterator I = cgen_state_->row_func_->arg_begin(), E = cgen_state_->row_func_->arg_end();
+         I != E;
+         ++I) {
+      row_process_arg_types.push_back(I->getType());
+    }
+
+    for (auto& element : cgen_state_->defined_literals_) {
+      for (auto value : element.second) {
+        row_process_arg_types.push_back(value->getType());
+      }
+    }
+
+    auto ft = llvm::FunctionType::get(get_int_type(32, cgen_state_->context_), row_process_arg_types, false);
+    auto row_func_hoisted = llvm::Function::Create(
+        ft, llvm::Function::ExternalLinkage, "row_func_hoisted", cgen_state_->row_func_->getParent());
```

Comment:
Maybe give it a clearer name, e.g. `"row_func_hoisted_literals"` ?

Reply:
yes, makes sense